src/payment/strategies/moneris/moneris-payment-strategy.ts

```diff
@@ -49,22 +68,55 @@ export default class MonerisPaymentStrategy implements PaymentStrategy {
 
     async execute(payload: OrderRequestBody, options?: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
         const { payment , ...order } = payload;
-        const { paymentMethods: { getPaymentMethodOrThrow } } = this._store.getState();
 
         if (!payment) {
             throw new PaymentArgumentInvalidError(['payment']);
         }
 
-        const paymentMethod = getPaymentMethodOrThrow(payment.methodId);
-        const testMode = paymentMethod.config.testMode;
-
         const { isStoreCreditApplied: useStoreCredit } = this._store.getState().checkout.getCheckoutOrThrow();
 
         if (useStoreCredit !== undefined) {
             await this._store.dispatch(this._storeCreditActionCreator.applyStoreCredit(useStoreCredit));
         }
 
-        await this._store.dispatch(this._orderActionCreator.submitOrder(order, options));
+        try {
+            await this._store.dispatch(this._orderActionCreator.submitOrder(order, options));
+
+            if (payment.paymentData && isVaultedInstrument(payment.paymentData)) {
+                return await this.executeWithVaulted(payment);
+            }
+
+            return await this.executeWithCC(payment);
+        } catch (error) {
+            return Promise.reject(error);
+        }
+    }
+
+    finalize(): Promise<InternalCheckoutSelectors> {
+        return Promise.reject(new OrderFinalizationNotRequiredError());
+    }
+
+    deinitialize(): Promise<InternalCheckoutSelectors> {
+        if (this._windowEventListener) {
+            window.removeEventListener('message', this._windowEventListener);
+            this._windowEventListener = undefined;
+        }
+
+        if (this._iframe && this._iframe.parentNode) {
+            this._iframe.parentNode.removeChild(this._iframe);
+            this._iframe = undefined;
+        }
+
+        return Promise.resolve(this._store.getState());
+    }
+
+    protected async executeWithCC(payment: OrderPaymentRequestBody): Promise<InternalCheckoutSelectors> {
```

Comment:
Why is this a protected method and not a private? and can we please add a `_` as a prefix.